src/components/ObsEdit/ObsEdit.js

```diff
@@ -113,6 +118,34 @@ const ObsEdit = ( ): Node => {
 
   const currentObs = observations[currentObsIndex];
 
+  useEffect( () => {
+    if ( currentObs ) {
+      const fetchLocation = async () => {
+        const location = await fetchUserLocation( );
+
+        updateObservationKeys( {
+          place_guess: location?.place_guess,
+          latitude: location?.latitude,
+          longitude: location?.longitude,
+          positional_accuracy: location?.positional_accuracy
+        } );
+        if ( location && mounted ) {
```

Comment:
This works, though instead of `mounted` I think you could check for the presence of `currentObs`.